lib/traiderdaive/traiderdaive/gym_environments/simple_hyperdrive_env.py

```diff
@@ -0,0 +1,443 @@
+
+        # How to render the environment
+        # TODO figure out what this does
+        render_mode: str | None = None
+
+        # RL Bot Config
+        # The constant trade amounts for longs and shorts
+        rl_agent_budget: FixedPoint = FixedPoint(1_000_000)
+        trade_base_amount: FixedPoint = FixedPoint(1000)
+        reward_scale: float = 1e-17
+        window_size: int = 10
+        episode_length: int = 200
+
+        # Other bots config
+        num_random_bots: int = 3
+        random_bot_budget: FixedPoint = FixedPoint(1_000_000)
+
+    # Defines allowed render modes and fps
+    metadata = metadata = {"render_modes": ["human"], "render_fps": 4}
+
+    def __init__(
+        self,
+        gym_config: Config,
+    ):
+        """Initializes the environment"""
+        # TODO parameterize these in the gym config
+        local_chain_config = LocalChain.Config(block_timestamp_interval=3600)
+        self.chain = LocalChain(local_chain_config)
+        initial_pool_config = InteractiveHyperdrive.Config()
+        self.interactive_hyperdrive = InteractiveHyperdrive(self.chain, initial_pool_config)
+
+        # Define the rl bot
+        self.rl_bot = self.interactive_hyperdrive.init_agent(base=gym_config.rl_agent_budget, name="rl_bot")
+        # Define the random bots
+        self.random_bots = [
+            self.interactive_hyperdrive.init_agent(
+                base=gym_config.random_bot_budget,
+                policy=PolicyZoo.random,
+                # TODO set the seed per random bot here for reproducability
+                policy_config=PolicyZoo.random.Config(),
+                name="random_bot_" + str(i),
+            )
+            for i in range(gym_config.num_random_bots)
+        ]
+
+        # Save a snapshot of initial conditions for resets
+        self.chain.save_snapshot()
+
+        assert gym_config.render_mode is None or gym_config.render_mode in self.metadata["render_modes"]
+        self.render_mode = gym_config.render_mode
+
+        self.gym_config = gym_config
+
+        # This action space encompasses buy and sell of bonds based on the current taken position
+        # Following https://github.com/AminHP/gym-anytrading/tree/master
+        # This binary action space with a position state variable results in one of the 4 following trades:
+        # - If the current position is short and the agent's action is to buy, the environment will
+        #     close the short position and open a long position. The current position is then set to long.
+        # - If the current position is long and the agent's action is to buy, this is noop (i.e., hold long position)
+        # - If the current position is long and the agent's action is to sell, the environment will
+        #   close the long position and open a short position. The current position is then set to short.
+        # - If the current position is short and the agent's action is to sell, this is noop (i.e., hold short position)
```

Comment:
opening a short is selling bonds, closing a short is buying bonds, opening a long is buying bonds, closing a long is selling bonds. 

Is that right?

What's your reasoning behind setting the action types to buy & sell instead of open & close?